pkg/forwarder/transaction.go

```diff
@@ -20,17 +22,61 @@ import (
 )
 
 var (
-	transactionsRetryQueueSize = expvar.Int{}
-	transactionsSuccessful     = expvar.Int{}
-	transactionsDroppedOnInput = expvar.Int{}
-	transactionsErrors         = expvar.Int{}
+	transactionsRetryQueueSize       = expvar.Int{}
+	transactionsSuccessful           = expvar.Int{}
+	transactionsDroppedOnInput       = expvar.Int{}
+	transactionsErrors               = expvar.Int{}
+	transactionsErrorsByType         = expvar.Map{}
+	transactionsDNSErrors            = expvar.Int{}
+	transactionsTLSErrors            = expvar.Int{}
+	transactionsConnectionErrors     = expvar.Int{}
+	transactionsWroteRequestErrors   = expvar.Int{}
+	transactionsCreatedRequestErrors = expvar.Int{}
+	transactionsSentRequestErrors    = expvar.Int{}
+	transactionsHTTPErrors           = expvar.Int{}
 )
 
+var trace = &httptrace.ClientTrace{
+	DNSDone: func(dnsInfo httptrace.DNSDoneInfo) {
+		if dnsInfo.Err != nil {
+			transactionsDNSErrors.Add(1)
+			log.Debugf("DNS Lookup failure: %s", dnsInfo.Err)
+		}
+	},
+	WroteRequest: func(wroteInfo httptrace.WroteRequestInfo) {
+		if wroteInfo.Err != nil {
+			transactionsWroteRequestErrors.Add(1)
+			log.Debugf("Request writing failure: %s", wroteInfo.Err)
+		}
+	},
+	ConnectDone: func(network, addr string, err error) {
+		if err != nil {
+			transactionsConnectionErrors.Add(1)
+			log.Debugf("Connection failure: %s", err)
+		}
+	},
+	TLSHandshakeDone: func(tlsState tls.ConnectionState, err error) {
+		if err != nil {
+			transactionsTLSErrors.Add(1)
+			log.Errorf("TLS Handshake failure: %s", err)
+		}
+	},
+}
+
 func initTransactionExpvars() {
+	transactionsErrorsByType.Init()
 	transactionsExpvars.Set("RetryQueueSize", &transactionsRetryQueueSize)
 	transactionsExpvars.Set("Success", &transactionsSuccessful)
 	transactionsExpvars.Set("DroppedOnInput", &transactionsDroppedOnInput)
 	transactionsExpvars.Set("Errors", &transactionsErrors)
+	transactionsExpvars.Set("ErrorsByType", &transactionsErrorsByType)
```

Comment:
Let's update https://github.com/DataDog/datadog-agent/blob/master/cmd/agent/dist/conf.d/go_expvar.d/agent_stats.yaml.example
with the new expvars